src/ansiblelint/rules/pattern.md

```diff
@@ -0,0 +1,3 @@
+# pattern
+
+This rule aims to validate Ansible pattern directory structure.
```

Comment:
- all the sub-rules (specific messages) need to documented, like we have on other rules.

if I remember correctly a missing `README.md` file should generate an error. I suspect that one named `pattern[missing-readme]` would be in this case.